packages/state-transition/src/cache/epochCache.ts

```diff
@@ -549,6 +571,20 @@ export class EpochCache {
     // ```
     this.epoch = computeEpochAtSlot(state.slot);
     this.syncPeriod = computeSyncPeriodAtEpoch(this.epoch);
+
+    // To populate finalized cache and prune unfinalized cache with validators that just entered the activation queue
+    if (this.isAfterEIP6110()) {
+      const expectedActivationEligibilityEpoch = prevEpoch + 1;
+      const validators = state.validators;
+      for (const index of epochTransitionCache.indicesEligibleForActivationQueue) {
+        const validator = validators.getReadonly(index);
+        if (validator.activationEligibilityEpoch == expectedActivationEligibilityEpoch) {
```

Comment:
If the first validator in the `indicesEligibleForActivationQueue` has `validator.activationEligibilityEpoch < expectedActivationEligibilityEpoch`, this loop will break out. Is that the intended logic?

Reply:
> If the first validator in the `indicesEligibleForActivationQueue` has `validator.activationEligibilityEpoch < expectedActivationEligibilityEpoch`, this loop will break out. Is that the intended logic?

On each `process_epoch()`, `indicesEligibleForActivationQueue` starts off with empty list, and validators that are later added to it will be set `validator.activationEligibilityEpoch = currEpoch + 1` per [here](https://github.com/ChainSafe/lodestar/blob/3a6702e95401a930099b304e4462ca8f09b67b15/packages/state-transition/src/epoch/processRegistryUpdates.ts#L35).

I believe the entire if statement is not necessary. Will remove